lua/lspconfig/puppet.lua

```diff
@@ -0,0 +1,26 @@
+local configs = require 'lspconfig/configs'
+local util = require 'lspconfig/util'
+
+local server_name = "puppet"
+local bin_name = "puppet-languageserver"
+
+configs[server_name] = {
+  default_config = {
+    cmd = {bin_name, "--stdio"};
+    filetypes = {"puppet"};
+    root_dir = util.root_pattern("manifests", ".puppet-lint.rc", "hiera.yaml", ".git");
```

Comment:
can the server also handle falling back to util.path.dirname(fname)?  see pyright